lib/auth/permissions.go

```diff
@@ -99,14 +101,19 @@ func (a *authorizer) Authorize(ctx context.Context) (*Context, error) {
 		return nil, trace.AccessDenied("missing authentication context")
 	}
 	userI := ctx.Value(ContextUser)
-	authContext, err := a.fromUser(ctx, userI)
+	authContext, lockTargets, err := a.fromUser(ctx, userI)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Enforce applicable locks in force.
+	lock := a.lockWatcher.FindLockInForce(lockTargets...)
+	if lock != nil {
+		return nil, trace.AccessDenied(services.LockInForceMessage(lock))
+	}
```

Comment:
The situation's a bit more complicated since the lock targets may involve roles prior to role mapping (i.e. from `authContext.UnmappedIdentity`). There is a way to simplify this, I had it prepared for a future PR but I'll include it here.